File: Demo/Sources/ViewControllers/DataView/List/DataCells/ListDataViewCell.m
//
//  Globus
//

#import "ListDataViewCell.h"
#import "ListDataProvider.h"

@interface ListDataViewCell ()

@property(nonatomic, weak) IBOutlet UILabel* displayLabel;

@end

@implementation ListDataViewCell

- (CGSize)sizeForAvailableSize:(CGSize)size {
    return CGSizeMake(size.width, 100.0f);
}

- (void)setup {
    [super setup];
    
    _displayLabel.textColor = UIColor.darkGrayColor;
}

- (void)willShow {
    ListDataProviderModel* viewModel = self.item.data;
    _displayLabel.text = viewModel.title;
}

@end

File: Demo/Sources/ViewControllers/DataView/List/DataCells/ListDataViewCell.swift
//
//  Globus
//

import Globus

class ListDataViewCell: GLBDataViewCell {
    
    // MARK: - Outlet property
    
    @IBOutlet fileprivate weak var displayLabel: UILabel!
    
    // MARK: - Public func

    public override func setup() {
        super.setup();
        
        self.displayLabel.textColor = UIColor.darkGray
    }
    
    public override func size(forAvailableSize size: CGSize) -> CGSize {
        return CGSize.init(width: size.width, height: 88.0)
    }
    
    public override func willShow() {
        if let viewModel = self.item!.data as? ListDataViewModel {
            self.displayLabel.text = viewModel.title;
        }
    }
    
    // MARK: - GLBNibExtension
    
    public override class func nibName() -> String {
        return "ListDataViewCell-Swift"
    }
    
}
